Add support for SOURCE_DATE_EPOCH to avoid embedding timestamp in .jar
file.

https://reproducible-builds.org/docs/source-date-epoch/

Index: apiguardian/build.gradle
===================================================================
--- apiguardian.orig/build.gradle
+++ apiguardian/build.gradle
@@ -8,7 +8,14 @@ plugins {
 	id 'signing'
 }
 
-Date buildTimeAndDate = new Date()
+// https://reproducible-builds.org/docs/source-date-epoch/
+String source_date_epoch = System.getenv("SOURCE_DATE_EPOCH");
+if (source_date_epoch != null) {
+   TimeZone.setDefault(TimeZone.getTimeZone("UTC"))
+}
+Date buildTimeAndDate = source_date_epoch == null ?
+    new Date() :
+    new Date(1000 * Long.parseLong(source_date_epoch))
 ext {
 	buildDate = new SimpleDateFormat('yyyy-MM-dd').format(buildTimeAndDate)
 	buildTime = new SimpleDateFormat('HH:mm:ss.SSSZ').format(buildTimeAndDate)
